### PR TITLE
Use Props factory methods in getting started tutorial

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
ah, `java.lang.IllegalArgumentException: erased Creator types are unsupported, use Props.create(actorClass, creator) instead`

I'll fix

**Comment by patriknw** [comment]
This makes me wonder why we keep the `def create[T <: Actor](creator: Creator[T]): Props`, since it can't be used with lambdas and I can't imagine someone crafting a `Creator` themselves nowadays and if so the additional class parameter wouldn't hurt much.

**Comment by patriknw** [comment]
Also deprecates Props.create without actorClass parameter 

Refs #25718

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-actor-tests/src/test/java/akka/actor/ActorCreationTest.java [hunk 1]
```diff
@@ -201,7 +201,7 @@ public void testWrongErasedStaticCreator() {
       Props.create(new G());
       assert false;
     } catch (IllegalArgumentException e) {
-      assertEquals("erased Creator types are unsupported, use Props.create(actorClass, creator) instead", e.getMessage());
+      assertEquals("erased Creator types (e.g. lambdas) are unsupported, use Props.create(actorClass, creator) instead", e.getMessage());
```
Comment (chbatey) [inline akka-actor-tests/src/test/java/akka/actor/ActorCreationTest.java]: :+1:

File: akka-actor-tests/src/test/java/akka/actor/JavaAPI.java [hunk 2]
```diff
@@ -77,9 +69,13 @@ public Object create() {
     });
   }
 
+  public static Props mkPropsWithLambda() {
+    return Props.create(JavaAPITestActor.class, JavaAPITestActor::new);
```
Comment (chbatey) [inline akka-actor-tests/src/test/java/akka/actor/JavaAPI.java]: I see no Lambda :)

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-actor-tests/src/test/java/akka/actor/JavaAPI.java (hunk 2 above)
Comment (patriknw) [reply akka-actor-tests/src/test/java/akka/actor/JavaAPI.java]: isn't method references just sugar for lambdas?

Comment (dwijnand) [reply akka-actor-tests/src/test/java/akka/actor/JavaAPI.java]: Java pulled the cotton candy over Chris' eyes.

**Pull request merged**